### PR TITLE
feat(python): add ast-based helpers for loops

### PR DESCRIPTION
**Pull request by Dario-DC**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

### CODE REVIEW

File: python/py_helpers.py [hunk 1]
```diff
@@ -188,3 +188,78 @@ def _find_if_bodies(tree):
...
+
+            return [tree.body] + [tree.orelse]
+
+        return [Node(ast.Module(body, [])) for body in _find_for_bodies(self.tree)]
+    
+    def find_if(self, if_str):
+        if_list = self._find_all(ast.If)
+        for if_statement in if_list:
+            if if_statement.find_conditions()[0].is_equivalent(if_str):                
+                return if_statement
```
Comment (ojeytonwilliams) [inline python/py_helpers.py:252]:
```suggestion
                return if_statement
        return Node()
```
Implicitly returning `None` breaks the chain and continued chaining gives you `AttributeError: 'NoneType' object has no attribute...` errors.

File: python/py_helpers.py [hunk 2]
```diff
@@ -188,3 +188,78 @@ def _find_if_bodies(tree):
...
+            
+    def find_while(self, while_str):
+        while_list = self._find_all(ast.While)
+        for while_loop in while_list:
+            if while_loop.find_while_conditions()[0].is_equivalent(while_str):                
+                return while_loop
```
Comment (ojeytonwilliams) [inline python/py_helpers.py:258]:
```suggestion
                return while_loop
        return Node()
```

File: python/py_helpers.py [hunk 3]
```diff
@@ -188,3 +188,78 @@ def _find_if_bodies(tree):
...
+            
+    def find_for(self, target_str, iter_str):
+        for_list = self._find_all(ast.For)
+        for for_loop in for_list:
+            if for_loop.find_for_vars().is_equivalent(target_str) \
+                and for_loop.find_for_iter().is_equivalent(iter_str):                
+                return for_loop
```
Comment (ojeytonwilliams) [inline python/py_helpers.py:265]:
```suggestion
                return for_loop
        return Node()
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
These are looking great, @Dario-DC, just some small issues.

Also, if you're happy with the suggestions, could you add the new features to the docs?

**Comment by Dario-DC** [comment]
> Also, if you're happy with the suggestions, could you add the new features to the docs?

Sure, I'm going to do add them to the docs @ojeytonwilliams

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1: 

Thanks again, @Dario-DC. I'm very happy to get these in.

**Pull request merged**